### ISSUE TITLE
The training procedure

### ISSUE THREAD

**Issue by zshyang** [facebookresearch/meshrcnn#31]
Hi,
I tried to train the network with data from `pix3d`. I saw the training metrics showing up during the training. What about validation? We want to use validation data to monitor the training procedure to prevent overfitting, right? But it seems that you only did the test on the test dataset at the end of the training.
Best regards,
zshyang.

**Comment by gkioxari** [comment]
Hi @zshyang 
The Pix3D dataset is **really small** and very class imbalanced. We decided not to split the dataset into train/val/test as this would make all splits even smaller which would lead to statistically insignificant analysis. Instead we provide two train/test splits. We used `S1` to tune the few hyper parameters of Mesh R-CNN and we used `S2` as the hidden split on which we never did any cross validation.

The only hyper parameter I tuned for Mesh R-CNN on Pix3D were the weight losses (for chamfer and edge length). The training schedule and other parameters were decided by me based on my experience with training neural networks all these years. This means that they can be totally wrong and more tuning can lead to better performance!

**Comment by gkioxari** [comment]
To elaborate a little more on the point of tuning on tiny datasets so that it's not misinterpreted. My point is that tuning to squeeze two or three points on a dataset that is very small like Pix3D (e.g. look at Pix3D test set, some categories have less than 20 instances in them), has no value for the advancement of AI or 3D or anything. Pix3D is a great dataset, but it's small and it definitely does not transfer to real world dataset like COCO or ImageNet, which is the real goal when solving for 3D shape inference. I get that conferences demand state-of-the-art performance, but squeezing two or three points by tuning on a small dataset is not what's going to bring the break through here. Instead, I recommend using Pix3D as a *proof of concept*, where you show that your novel idea works. If I ever review a paper that has results on Pix3D, I will not just look for these extra 3 points on top of Mesh R-CNN. I will look for the novelty in your approach and I will get excited if your idea has any hope of working on any real-world images. 

Ok, that marks the end of my thoughts this fine Tuesday evening!

**Comment by gkioxari** [comment]
Omg it's Wednesday. This quarantine is not working well for me apparently.

**Comment by zshyang** [comment]
Hi @gkioxari 
Thanks a lot! Your answer always solved my question.
